lib/azure/armrest/storage/managed_storage_helper.rb

```diff
@@ -49,37 +50,13 @@ module Azure::Armrest::Storage::ManagedStorageHelper
   #   p data.headers
   #   File.open('vm.vhd', 'a'){ |fh| fh.write(data.body) }
   #
-  def get_blob_raw(disk_name, resource_group = configuration.resource_group, options = {})
-    validate_resource_group(resource_group)
-
-    post_options = {
-      :access            => 'read',                    # Must be 'read'
-      :durationInSeconds => options[:duration] || 3600 # 1 hour default
-    }
-
-    # This call will give us an operations URL in the headers.
-    begin_get_access_url = build_url(resource_group, disk_name, 'BeginGetAccess')
-    begin_get_access_response = rest_post(begin_get_access_url, post_options.to_json)
-
-    headers = Azure::Armrest::ResponseHeaders.new(begin_get_access_response.headers)
-    status = wait(headers, 120, 5)
-
-    unless status.casecmp('succeeded').zero?
-      msg = "Unable to obtain an operations URL for #{disk_name}/#{resource_group}"
-      log('debug', "#{msg}: #{begin_get_access_response.headers}")
-      raise Azure::Armrest::NotFoundException.new(begin_get_access_response.code, msg, begin_get_access_response.body)
-    end
-
-    # Get the SAS URL from the BeginGetAccess call
-    op_url = headers.try(:azure_asyncoperation) || headers.location
-
-    # Dig the URL + SAS token URL out of the response
-    response = rest_get(op_url)
-
-    body = Azure::Armrest::ResponseBody.new(response.body)
-    sas_url = body.properties.output.access_sas
+  def open(disk_name, resource_group = configuration.resource_group, options = {})
+    sas_url = access_token(disk_name, resource_group, options)
+    ManagedDisk.new(self, disk_name, resource_group, sas_url)
+  end
 
-    # The same restrictions that apply to the StorageAccont method also apply here.
+  def read(sas_url, options = {})
```

Comment:
No need to pass `options` into `read`, use `@open_options`

Reply:
No - the options have the read offset and length.  This changes for every call to read.